assets/src/edit-story/components/panels/textStyle/padding.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback } from 'react';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import clamp from '../../../utils/clamp';
+import { Label, Row, Numeric, Toggle, usePresubmitHandler } from '../../form';
+import { Lock, Unlock } from '../../../icons';
+import { useCommonObjectValue } from '../utils';
+
+const DEFAULT_PADDING = { horizontal: 0, vertical: 0, locked: true };
+
+const MIN_MAX = {
+  PADDING: {
+    MIN: 0,
+    MAX: 300,
+  },
+};
+
+const BoxedNumeric = styled(Numeric)`
+  padding: 6px 6px;
+  border-radius: 4px;
+
+  ${({ stretch }) => stretch && `flex-basis: auto;`}
+`;
+
+const Space = styled.div`
+  flex: 0 0 10px;
+`;
+
+function PaddingControls({ selectedElements, pushUpdateForObject }) {
+  const padding = useCommonObjectValue(
+    selectedElements,
+    'padding',
+    DEFAULT_PADDING
+  );
+
+  // Only if true for all selected elements, will the toggle be true
+  // (Note that this behavior is different from aspect lock ratio)
+  const lockPadding = padding.locked === true;
+
+  const handleChange = useCallback(
+    (newPadding, submit = false) => {
+      pushUpdateForObject('padding', newPadding, DEFAULT_PADDING, submit);
+    },
+    [pushUpdateForObject]
+  );
+
+  usePresubmitHandler(
+    ({ padding: { horizontal, vertical, ...rest } }) => ({
+      padding: {
+        ...rest,
+        horizontal: clamp(horizontal, MIN_MAX.PADDING),
+        vertical: clamp(vertical, MIN_MAX.PADDING),
+      },
+    }),
+    []
+  );
+
+  const firstInputProperties = lockPadding
```

Comment:
Curious, what was the reason for combining the padding files? DRY? Or accessibility reasons, too?
It's reducing the readability a bit but perhaps the benefits have more weight.

Reply:
Accessibility - I wanted keyboard focus to stay on the lock icon button when clicking it (so when tabbing through and pressing enter on the button, you could keep tabbing from that same button). Thus it had to be a single component and had to return the same object, but with slightly different properties for the button, so React would treat it as the same DOM element and the browser would not move focus around.

I actually first tried manually refocusing the "new" button, but it was horrible.